samples/cellular/nrf_cloud_multi_service/src/cloud_connection.c

```diff
@@ -223,6 +244,14 @@ static bool connect_cloud(void)
 	/* Connect via CoAP -- blocking. */
 	err = nrf_cloud_coap_connect(CONFIG_APP_VERSION);
 
+	/* Connection was paused, so resume it. */
+	if (err > 0) {
+		err = nrf_cloud_coap_resume();
+		if (err) {
+			disconnect_cloud();
```

Comment:
Calling disconnect_cloud here really increases the difficulty of understanding the sample control flow, IMO

What is the reason for calling it, and could the same be achieved another way?

Reply:
The resume failed, so we will need to do a full handshake.

Reply:
Actually, there is a better way.